plugins/modules/katello_repository_set.py

```diff
@@ -140,68 +132,68 @@ def get_desired_repos(desired_substitutions, available_repos):
     return desired_repos
 
 
-def repository_set(module, name, organization, product, label, state, repositories=[]):
-    changed = False
-    organization = find_organization(module, organization)
-    if product:
-        product = find_product(module, product, organization)
-    repo_set = find_repository_set(module, name=name, product=product, organization=organization, label=label)
+def main():
+    module = KatelloEntityAnsibleModule(
+        argument_spec=dict(
+            name=dict(default=None),
+            product=dict(default=None),
+            label=dict(default=None),
+            repositories=dict(required=True, type='list'),
```

Comment:
Should this use `type='nested_list'` ?

Reply:
`nested_list` is only really useful when we actually use the `entity_spec`, but this module doesn't because the `repository_set` API is very katello-like ;)

Reply:
Did you mean to use Ansible's own `elements='dict'`, and `options=…` (as done by `nested_list`)

Reply:
I've set [`elements='dict'`](https://docs.ansible.com/ansible/latest/dev_guide/developing_program_flow_modules.html#elements) now,